engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,271 @@
+            return filteredIndex.isNonempty();
+        }
+
+        @Override
+        public Boolean getPrevBoolean(long rowKey) {
+            WritableRowSet filteredIndex = filter.filter(RowSetFactory.fromKeys(rowKey), rowSet, tableToFilter, true);
+            return filteredIndex.isNonempty();
+        }
+
+        @Override
+        public Object get(final long rowKey) {
+            return getBoolean(rowKey);
+        }
+
+        @Override
+        public Object getPrev(final long rowKey) {
+            return getPrevBoolean(rowKey);
+        }
+
+        @Override
+        public ChunkType getChunkType() {
+            return ChunkType.Object;
+        }
+
+        @Override
+        public FillContext makeFillContext(final int chunkCapacity) {
+            return FILL_CONTEXT_INSTANCE;
+        }
+
+        @Override
+        public void fillChunk(
+                @NotNull final FillContext fillContext,
+                @NotNull final WritableChunk<? super Values> destination,
+                @NotNull final RowSequence rowSequence) {
+            doFill(rowSequence, destination, false);
+        }
+
+        @Override
+        public void fillPrevChunk(
+                @NotNull final FillContext fillContext,
+                @NotNull final WritableChunk<? super Values> destination,
+                @NotNull final RowSequence rowSequence) {
+            doFill(rowSequence, destination, true);
+        }
+
+        private void doFill(@NotNull RowSequence rowSequence, WritableChunk<? super Values> destination,
+                boolean usePrev) {
+            final WritableObjectChunk<Boolean, ?> booleanDestination = destination.asWritableObjectChunk();
+            booleanDestination.setSize(rowSequence.intSize());
+            try (final RowSet inputRowSet = rowSequence.asRowSet();
+                    final RowSet filtered = filter.filter(inputRowSet, rowSet, tableToFilter, usePrev);
+                    final RowSet.Iterator inputIt = inputRowSet.iterator();
+                    final RowSet.Iterator trueIt = filtered.iterator()) {
+                long nextTrue = trueIt.hasNext() ? trueIt.nextLong() : -1;
+                int offset = 0;
+                while (nextTrue >= 0) {
+                    // the input iterator is a superset of the true iterator, so we can always find out what
+                    // the next value is without needing to check hasNext
+                    final long nextInput = inputIt.nextLong();
+                    final boolean found = nextInput == nextTrue;
+                    booleanDestination.set(offset++, found);
+                    if (found) {
+                        nextTrue = trueIt.hasNext() ? trueIt.nextLong() : -1;
+                    }
+                }
+                // fill everything else up with false, because nothing else can match
+                booleanDestination.fillWithBoxedValue(offset, booleanDestination.size() - offset, false);
```

Comment:
Prefer bulk advance?
```suggestion
                    final RowSequence.Iterator inputRows = inputRowSet.getRowSequenceIterator();
                    final RowSet.Iterator trueRows = filtered.iterator()) {
                int offset = 0;
                while (trueRows.hasNext()) {
                    final long nextTrue = trueRows.nextLong();
                    // Find all the false rows between the last consumed input row and the next true row
                    final int falsesSkipped = (int) inputRows.advanceAndGetPositionDistance(nextTrue + 1) - 1;
                    if (falsesSkipped > 0) {
                        booleanDestination.fillWithValue(offset, falsesSkipped, false);
                        offset += falsesSkipped;
                    }
                    booleanDestination.set(offset++, true);
                }
                final int remainingFalses = booleanDestination.size() - offset;
                // Fill everything else up with false, because we've exhausted the trues
                if (remainingFalses > 0) {
                    booleanDestination.fillWithValue(offset, remainingFalses, false);
                }
```

Reply:
Original code (density is the fraction of true values, uniformly distributed):
```
Density: 1.0000, Nanos: 11560028500, per cell=11.56
Density: 0.9999, Nanos: 13277620834, per cell=13.28
Density: 0.9990, Nanos: 14978527958, per cell=14.98
Density: 0.8725, Nanos: 21772909750, per cell=21.77
Density: 0.7500, Nanos: 28369796167, per cell=28.37
Density: 0.5000, Nanos: 36947907333, per cell=36.95
Density: 0.2500, Nanos: 24845209958, per cell=24.85
Density: 0.1250, Nanos: 17964264291, per cell=17.96
Density: 0.0010, Nanos: 11357967500, per cell=11.36
Density: 0.0001, Nanos: 10771053167, per cell=10.77
```

New code:
```
Density: 1.0000, Nanos: 11589953167, per cell=11.59
Density: 0.9999, Nanos: 13430658834, per cell=13.43
Density: 0.9990, Nanos: 15374023083, per cell=15.37
Density: 0.8725, Nanos: 22405263417, per cell=22.41
Density: 0.7500, Nanos: 28906296084, per cell=28.91
Density: 0.5000, Nanos: 36402417875, per cell=36.40
Density: 0.2500, Nanos: 24535559166, per cell=24.54
Density: 0.1250, Nanos: 17518296583, per cell=17.52
Density: 0.0010, Nanos: 10686682416, per cell=10.69
Density: 0.0001, Nanos: 10635898708, per cell=10.64
```

Reply:
I think I like the new code better, but there's no strong case for either.